lib/client.js

```diff
@@ -19,6 +19,23 @@ function validateKey(key, operation) {
     misc.assert(key.length < 250, 'Key must be less than 250 characters long');
 }
 
+function validateNamespaceKey(key, operation) {
+    validateKey(key, operation);
+    misc.assert(key.length < 230, 'Key must be less than 230 characters long when using namespaces');
```

Comment:
So it's good that you're checking for the namespace key to be shorter, but this without a check on the length of namespace itself isn't super useful. There is nothing preventing me from having a `key` that is 229 characters and a `namespace` that is also 229 characters and being far past the 250 key limit set by Memcached core.

I'd much prefer to either

1. Add a limit on the length of a namespace as well (here at 20 characters)
2. Just run `validateKey()` after appending the namespace with the key

I favor 2. because it doesn't establish an arbitrary limit on namespace (you've chosen 20 and that seems rather arbitrary).

Reply:
As I wrote in my other comment, I think the idea of this feature was misunderstood.
It doesn't matter what the length of the name of the namespace is. The prefix is always going to be a unix timestamp which is around 13 characters. The key which uses namespace needs to be at most (250-13) characters long.

Reply:
Great works for adding the name spacing feature!
https://www.npmjs.com/package/memcached-lite has a similar feature named `keyPrefix`.
I use it with some prefix string longer than 20 characters.
In my case, the prefix would include some from below:
- `require("./package.json").name` - this works when a memcached server is shared by multiple apps.
- `require("./package.json").version` - this works when updating the app which has a difference schema than before
- `__filename.split("/").pop()` - this works to separate namespaces for each files.
- `+new Date()` as @royshouvik mentioned